### PR TITLE
[Chat] Chat Message Avatar

### PR DESCRIPTION
**Pull request by tangziyi001**
1. Fetch profile info in chat room page so profile image can be displayed as avatar
2. Slight refactor of user profile builder to make it more adaptable to other use cases (main page, chat page, and hopefully setting preview page)
3. Add download image helper function to conveniently download UIImage
4. Unify getURL function in different places

Screen Shot 2019-03-17 at 4 52 26 PM
Screen Shot 2019-03-17 at 4 52 31 PM

### CODE REVIEW

File: NewDawn/NewDawn/HttpUtil/HttpUtil.swift [hunk 1]
```diff
@@ -89,11 +82,18 @@ class HttpUtil{
         task.resume()
     }
     
-    static func getURL(path:String, prod:Bool = CONNECT_TO_PROD) -> URL {
+    static func getURL(path:String, prod:Bool = CONNECT_TO_PROD, isMedia: Bool = false) -> URL {
         var final_path = path
         if final_path.hasPrefix("/") == false {
             final_path = "/" + path
         }
+        if isMedia == true {
```
Comment (duckmoll) [inline NewDawn/NewDawn/HttpUtil/HttpUtil.swift]: Can you explain more about your usage of "isMedia"?

Comment (tangziyi001) [reply NewDawn/NewDawn/HttpUtil/HttpUtil.swift]: by default, media URLs are not prepended with /api/v1/. So I split them into two different paths

Comment (duckmoll) [reply NewDawn/NewDawn/HttpUtil/HttpUtil.swift]: I see, but aren't all the current image uploading started with "/api/v1/" ?

Comment (tangziyi001) [reply NewDawn/NewDawn/HttpUtil/HttpUtil.swift]: Image upload is a POST or Multipart API I assume, which starts with /api/v1/
Image download is directly fetching static resources from file system, which is beyond the regular api path: https://github.com/new-dawn/new_dawn_server/blob/master/new_dawn_server/settings.py#L138

Comment (duckmoll) [reply NewDawn/NewDawn/HttpUtil/HttpUtil.swift]: Got it. Thanks for explanation.

**Pull request merged**